apps/system/test/unit/settings_migrator_test.js

```diff
@@ -73,4 +75,41 @@ suite('system/settings_migrator', function() {
       assert.equal(window.navigator.mozSettings.mSettings[keyHour12], true);
     });
   });
+
+
+  suite('when old passcode is present', function() {
+    setup(function() {
+      window.navigator.mozSettings.mSetup();
+      var cset = {};
+      cset[oldPasscode] = '1234';
+      window.navigator.mozSettings.mSet(cset);
+
+      realLazyLoader = window.LazyLoader;
+      window.LazyLoader = MockLazyLoader;
+      this.sinon.spy(window.LazyLoader, 'load');
+
+      window.PasscodeHelper = {
+        set: function() {
+          return Promise.resolve();
+        }
+      };
+      this.sinon.spy(window.PasscodeHelper, 'set');
+
+      window.settingsMigrator = new SettingsMigrator();
+      window.settingsMigrator.start();
+      clock.tick(50);
+    });
+
+    test('it uses passcodehelper to set new.', function() {
+        assert.ok(window.LazyLoader.load.called, 'should use lazyloader');
+        assert.ok(window.PasscodeHelper.set.called, 'should use' +
+          ' PasscodeHelper');
+    });
+
+    teardown(function() {
+      window.LazyLoader = realLazyLoader;
+    });
+
```

Comment:
please also add suite('when old passcode is not present') test case to make sure its not run